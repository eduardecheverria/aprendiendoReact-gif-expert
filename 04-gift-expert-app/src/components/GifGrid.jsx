import useFetchGifs from '../hooks/useFetchGifs';
import '../styles.css';
import GifItem from './GifItem';
const GifGrid = ({ category }) => {
    const { images, isLoading } = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>
            {isLoading && <h2>Cargando...</h2>}
            <ol className="card-grid">
                {images.map((image) => (
                    <GifItem key={image.id} {...image} />
                ))}
            </ol>
        </>
    );
};

export default GifGrid;
