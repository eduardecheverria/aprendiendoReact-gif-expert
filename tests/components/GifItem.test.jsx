import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas en GifItem', () => {
    const title = 'saitama';
    const url = 'https://one-punch.com/hola.jpg';

    test('Debe hacer match con el snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('Deebe de tener el alt y el titulo indicado', () => {
        render(<GifItem title={title} url={url} />);
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('Debe mostrar el titulo en el componente', () => {
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    });
});
