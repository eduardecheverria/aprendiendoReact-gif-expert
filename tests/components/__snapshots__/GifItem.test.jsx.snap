// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="saitama"
      src="https://one-punch.com/hola.jpg"
    />
    <p>
      saitama
    </p>
  </div>
</div>
`;
